src/leappto/providers/libvirt_provider.py

```diff
@@ -125,9 +133,41 @@ def __inspect_os(domain_name):
 
             return (hostname, Installation(OperatingSystem(distro, '{}.{}'.format(major, minor)), packages))
 
-        def __get_agent_info(domain):
-            domxml = ET.fromstring(domain.XMLDesc(0))
-            channel = domxml.find('devices/channel/source')
+        def __vagrant_ssh_checkoutput(domain_name, command):
+            with open(os.devnull, 'w') as devnull:
+                cmd = ["vagrant", "ssh", domain_name, "-c", command]
+                return check_output(cmd, stderr=devnull)
+
+        def __get_ssh_info(domain_name):
+            domain_name, _ = domain_name.split('_', 1)
+            cmd = 'vagrant global-status | grep {} | cut -d\\  -f1'.format(domain_name)
+            domain_name = check_output(cmd, shell=True).strip()
+            cmd = """python -c 'import platform; print chr(0xa).join(platform.linux_distribution()[:2])'"""
```

Comment:
Ah, interesting. As an added bonus, using Python level checks like this will generalise nicely to arbitrary remote system access (including via Ansible, if/when we go that way).